### PR TITLE
Added new blog - DOM manipulation in JavaScript

### LINKED ISSUE #2017: [Feature Request]: Add new blog - DOM manipulation in javascript
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to add a new blog article on this topic, which explains dom manipulation ideas with simple examples.

### Use Case

NA

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
## Related Issue

fixes #2017 

## Description

Added new blog.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

### CODE REVIEW

File: blog/DOM manipulation in JavaScript.md [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+<p>This is a paragraph.</p>
+<div>
+    <p>Another paragraph inside a div.</p>
+    <p>Second paragraph inside a div.</p>
+</div>
+```
+
+```js
+// Get all <p> elements in the document
+var paragraphs = document.getElementsByTagName("p");
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:122]: Using `var` is generally not recommended. Prioritize `const` for variables that do not change. If you need to reassign a variable, use `let`. Avoid `var` as a matter of good practice.

Comment (Ajay-Dhangar) [reply blog/DOM manipulation in JavaScript.md:122]: So, replace `var` to `const`

File: blog/DOM manipulation in JavaScript.md [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+// Loop through and log the text content of each <p> element
+for (var i = 0; i < paragraphs.length; i++) {
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:125]: `var` to `let`

File: blog/DOM manipulation in JavaScript.md [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+    console.log(paragraphs[i].textContent);
+}
+```
+
+- The `querySelector` method in JavaScript allows you to select and retrieve the first element that matches a specified CSS selector within the document or within a specific element,
+
+```js
+// Select the first <p> element in the document
+var firstParagraph = document.querySelector("p");
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:134]: `var` to `const`

File: blog/DOM manipulation in JavaScript.md [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+
+// Select the element with id="main-title"
+var titleElement = document.querySelector("#main-title");
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:137]: `var` to `const`

File: blog/DOM manipulation in JavaScript.md [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+
+// Select the first element with class="intro"
+var introParagraph = document.querySelector(".intro");
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:140]: `var` to `const`

File: blog/DOM manipulation in JavaScript.md [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+
+// Select the first <p> element inside the <div>
+var paragraphInDiv = document.querySelector("div p");
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:143]: `var` to `const`

File: blog/DOM manipulation in JavaScript.md [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+```
+
+- The `querySelectorAll` method in JavaScript allows you to select and retrieve a list (or NodeList) of all elements that match a specified CSS selector within the document or within a specific element. Unlike `querySelector`, which returns only the first matching element, `querySelectorAll` returns a NodeList containing all matching elements.
+
+```js
+
+// Select all <p> elements in the document
+var paragraphs = document.querySelectorAll("p");
```
Comment (Ajay-Dhangar) [inline blog/DOM manipulation in JavaScript.md:151]: also, replace `var` to `const`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
### update your code for best practice 

Using `var` is not considered good practice. Instead, prioritize using `const` for variables that won't be reassigned. For variables that need to be reassigned, especially in loops, use `let`. Avoid using `var` to follow modern JavaScript best practices.

**Comment by DharshiBalasubramaniyam** [comment]
> ### update your code for best practice
> Using `var` is not considered good practice. Instead, prioritize using `const` for variables that won't be reassigned. For variables that need to be reassigned, especially in loops, use `let`. Avoid using `var` to follow modern JavaScript best practices.

done!

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**